src/rules/no-restricted-patterns.js

```diff
@@ -1,79 +1,115 @@
 var _ = require('lodash');
 var rule = 'no-restricted-patterns';
-var availableConfigs ={
+var availableConfigs = {
   'Global': [],
   'Scenario': [],
   'ScenarioOutline': [],
   'Background': [],
   'Feature': []
 };
 
-function allowedPatterns(feature, fileName, configuration) {
-  // Patterns applied to all features, backgrounds, etc.
+
+function noRestrictedPatterns(feature, fileName, configuration) {
+  var errors = [];
+
+  // Patterns applied to everything; feature, scenarios, etc.
   var globalPatterns = (configuration.Global || []).map(function(pattern) {
     return new RegExp(pattern, 'i');
   });
-  var featurePatterns = (configuration.Feature || []).map(function(pattern) {
-    return new RegExp(pattern, 'i');
+
+  var restrictedPatterns = {};
+  Object.keys(availableConfigs).forEach(function(type) {
+    restrictedPatterns[type] = (configuration[type] || []).map(function(pattern) {
+      return new RegExp(pattern, 'i');
+    });
+    restrictedPatterns[type] = restrictedPatterns[type].concat(globalPatterns);
   });
 
-  var featureErrors = checkFeature(feature, globalPatterns.concat(featurePatterns));
+  checkFeatureNode(feature, restrictedPatterns, errors);
+  return errors;
+}
 
-  var childrenErrors = _(feature.children).map(function(child) {
-    // Combine global patterns with the child type (Background, Scenario, etc).
-    var allowedPatterns = globalPatterns.concat(configuration[child.type] || []);
-    return checkSteps(child || [], allowedPatterns);
-  }).flatten().value();
 
-  return featureErrors.concat(childrenErrors);
+function checkFeatureNode(node, restrictedPatterns, errors) {
+  if (_.isEmpty(node)) {
+    return;
+  }
+  checkNameAndDescription(node, restrictedPatterns, errors);
+  node.children.forEach(function(child) {
+    checkFeatureChildNode(child, restrictedPatterns, errors);
+  });  
 }
 
-function checkFeature(feature, allowedPatterns) {
-  // Check title and description of the feature.
-  var name = feature.name || '';
-  var description = feature.description || '';
-  var errors = [];
-  allowedPatterns.forEach(function (pattern) {
-    if (name.match(pattern)) {
-      errors.push({message: 'Restricted pattern "' + name.trim() + '" on Feature',
-        rule: rule,
-        line: feature.location.line});
-    }
-    if (description.match(pattern)) {
-      errors.push({message: 'Restricted pattern "' + description.trim() + '" on Feature',
-        rule: rule,
-        line: feature.location.line});
-    }
-  });
-  return errors;
+
+function checkNameAndDescription(node, restrictedPatterns, errors) {
+  restrictedPatterns[node.type]
+    .forEach(function(pattern) {
+      check(node, 'name', pattern, errors);
```

Comment:
Scenarios and ScenarioOutlines also have a name
Scenarios and ScenarioOutlines and Background also have a description